syft/generic/frameworks/hook/hook_args.py

```diff
@@ -717,6 +717,9 @@ def register_tensor(
         response_ids: List of ids where the tensor should be stored
             and each id is pop out when needed.
     """
+    # `de_register` initially as this causes the original PT
```

Comment:
What causes the original PT not to get GC'd? Unclear from this comment.

Reply:
Updated :smile: